src/toniarts/openkeeper/view/PlayerCamera.java

```diff
@@ -29,54 +33,100 @@
 public class PlayerCamera {
 
     private final Camera cam;
-    private static final float ZOOM_SPEED = 25f;
-    private static final float MOVE_SPEED = 25f;
-    private static final float ROTATION_SPEED = 5f;
+    private final Thing.Camera presets;
+    private Vector2f limit;
 
-    public PlayerCamera(Camera cam) {
+    public PlayerCamera(Camera cam, Thing.Camera presets) {
         this.cam = cam;
+        this.presets = presets;
+        if (presets == null) {
+            throw new RuntimeException("Very bad");
+        }
+        initialize();
     }
 
-    protected void zoomCamera(float value, boolean mouse) {
+    /**
+     * Load the initial main menu camera position
+     */
+    private void initialize() {
+        // FIXME maybe another default angles
+        Quaternion q = new Quaternion().fromAngles(FastMath.PI * (0.5f - presets.getAnglePitch() / 1024f),
+                FastMath.PI * (0.75f + presets.getAngleYaw() / 1024f),
+                FastMath.PI * (0 + presets.getAngleRoll() / 1024f));
+
+        cam.setRotation(q);
+        cam.setLocation(new Vector3f(0, presets.getHeight(), 0));
+        // FIXME fov maybe have another formula. Need correct?
+        cam.setFrustumPerspective(presets.getFov() * FastMath.RAD_TO_DEG / 4, (float) cam.getWidth() / cam.getHeight(), 0.1f, 100f);
+    }
 
+    protected void zoom(float value) {
+        if (presets.getFlags().contains(Thing.Camera.CameraFlag.DISABLE_ZOOM)) {
+            return;
+        }
         // Zooming is here moving towards the looking at position
         Vector3f vel = new Vector3f();
         Vector3f pos = cam.getLocation().clone();
         cam.getDirection(vel);
-        vel.multLocal(mouse ? value : value * ZOOM_SPEED);

```

Comment:
PlayerCamera class only move camera and check boundaries. But PlayerCameraState class know what speed is setted.